### ISSUE TITLE
RFC: Prolific Contributor Badge/Certificate for Contributions.

### ISSUE THREAD

**Issue by raisedadead** [freeCodeCamp/freeCodeCamp#16876]
I think its a good idea to award a prolific contributor badge/certificate to the community members who have done a prolific job in making the platform what it is.

I understand that there are ways that does display there work in the list of contributors. But to drive the curriculum we should consider a badge/certificate of appreciation.

This should recognise the common effort in driving across all domains, not just coding the curriculum but anything and everything that helps drive the community.

We could use a flow of nomination and votes on the forum, and tie the award to their profile.

Thoughts?

**Comment by Ethan-Arrowood** [comment]
This sounds like a great idea! Would this be for FreeCodeCamp/FreeCodeCamp contributions only or for any FreeCodeCamp repository?

**Comment by mstellaluna** [comment]
@raisedadead I think its a great idea. My question is, how would this work for the nomination and votes because people from FCC Chat don't tend to go on to the forums and vice versa. We have one camper in chat that has over 9000 brownies from helping campers. 

Also, some people, such as myself, don't visibly contribute to FCC, its more of a behind the scenes.

**Comment by raisedadead** [comment]
Hi @Ethan-Arrowood It should be targeted to all domains of the community, including but not limited to codebase, forum, chat, youtube, etc., but the focus is towards an effort that increases the motivation to contribute back to the community and offer a recognition of such effort.

Hi @mstellaluna, I haven't thought much into the the workflow, and was hoping for inputs from the community, in this thread. I think could some form of non-conventional approach. 

**I do agree that some contributions like yours  in QA of the platform are so crucial in catching issues early on and often are known only to a few. I truly thank you for that. 

I don't have a concrete answer for that use case. However, I think we should be able to get ideas here.
The intent is to be able to encourage and reward this, in a way that inspires the community.

**Comment by mstellaluna** [comment]
@raisedadead agreed. let me think of some ideas :)

**Comment by mstellaluna** [comment]
@raisedadead Okay.. quick idea. 

How about each platform votes for its own since forum users aren't aware of chat users and vice versa. 

So a poll can be put on forums which we the chat moderators can post the link intermittently into the rooms for chat campers to go vote.

Forum platform votes for forum users.
Chat platform votes for chat users. 
Contributors room for the FCC Contributors (repo, FB, YT, medium etc)

**Comment by BerkeleyTrue** [comment]
I'd avoid anything that involves voting, as these are easily gamed and takes more work then they are worth.

Git commit lines contributed could be one metric.

> We have one camper in chat that has over 9000 brownies from helping campers.

That's amazing! we have the ability to distinguish between points from thanks and points from chalenges. We could use that as another metric (maybe another badge).

I'm not sure what would be a good metric for the forum.

**Comment by IsaacAbrahamson** [comment]
I like the idea, but the forums are a difficult one to judge.

Since I spend most of my time on the forum, I don't think it's fair that people on Gitter be rewarded with badges for giving quick little thanks while I could spend an hour writing a reply to someone and not even get one like. Not that I help people because of that, obviously, but if you are rewarding people, it needs to be consistent across all platforms, and that is hard because of the nuances different platforms have.

Here is a link that shows an overview of the users:
https://forum.freecodecamp.org/u

For example, take a look at the best users over the previous month. I have several problems with just basing rewards off of this. If you look at likes received, I think it is pretty accurate, but there are problems. For one, p1xt doesn't even visit our forums anymore, yet she is in the top 20 in likes this month. JABedford hasn't even been on the forum but 2 months, and he has been getting more likes than long-time members. If you consider the past week, jlave has gotten a ton of likes, just because he posted one job post. These got a job posts can get 40-50 likes easily, while more helpful posts by other campers may not even get any.

That's only looking at likes. If you look at replies, that's not fair either. bobbyc has more replies than me, yet I was one of the ones helping her with a wordpress problem. A lot of new users just create a ton of new posts instead of creating multiple replies in one post.

A voting system for the forum might be better, but there are still problems with that. It should be obvious that randeldawson, arielleslie, portablestick, and owel should be in the top campers. Yet, there are several people like me who still try to be helpful, we just aren't as pronounced. In a voting system, results would be biased. People who only recieved help from randeldawson would only vote for him, and other campers who do more behind the scenes and spend time in other categories like project feedback aren't going to be rewarded with votes even though they are critical to the success of the forum.

Just my thoughts.

**Comment by mstellaluna** [comment]
> Since I spend most of my time on the forum, I don't think it's fair that people on Gitter be rewarded with badges and 9000 brownies for giving quick little thanks while I could spend an hour writing a reply to someone and not even get one like.

I for one, I am deeply offended at this comment.  The campers in chat are, such myself, are there for a reason. Campers don't just get a one-line help. The 9000 point camper, is THE person to go to for javascript, and it not 'let me google that for you', its in-depth help. They get help for design, javascript, career advise, sometimes hey it happens we help with homework.  Don't assume we don't help in the chat, it is offensive to the gitter community as a whole.

**Comment by raisedadead** [comment]
Hey guys, thanks for the comments. We sincerely thank you for all the good work you do. 

I apologise for any understanding otherwise.

But, I strongly suggest lets be constructive about the feedback. I feel we should appreciate any effort (regardless of its quantum). And we should just keep the focus on discussing the best way to reward efforts.

I think metrics from the codebase contributions, is the only concrete reference we can award such badges. 

And for channels where biasing, can be involved, we can normalize metrics for the rewards. This is not a first time problem, for example, StackOverflow does this in a very immaculate way.

Lets brainstorm together and think of ideas.

**Comment by IsaacAbrahamson** [comment]
@mstellaluna 
> I for one, I am deeply offended at this comment. The campers in chat are, such myself, are there for a reason. Campers don't just get a one-line help. The 9000 point camper, is THE person to go to for javascript, and it not 'let me google that for you', its in-depth help. They get help for design, css, javascript, career advice, interview advice, sometimes hey it happens we help with homework. Don't assume we don't help in the chat, it is offensive to the gitter community as a whole.

I'm sorry that I came across as offensive or against the gitter community. I have nothing against anyone who helps out in any way big or small, and I greatly appreciate everyone who has gone out of their way to help me, especially when I first joined the Gitter chat.

My point was not that the Gitter community is not as helpful. I am sure that the camper with 9000 points is fabulous (as are the other campers who help out), and I'm thrilled that we have so many great people like that in our community.

I was trying to say that it is not fair to the many people on the forum who aren't often noticed for their help while someone on Gitter can be given a brownie point for saying "welcome" or "thanks". There is nothing wrong with Gitter having these rewards, but if people on Gitter are being rewarded because they are "Super Level 9000" campers, while people on the forum are only "level 400" campers even though they may work almost as hard for the community, that is a problem. I'm trying as best as I can to be constructive and share my perspective on an issue, and I'm not trying to criticize or put-down the Gitter community as inferior. I am just worried that since it seems easier to be easily rewarded on Gitter, people on Gitter will get many more badges than people on the forum who are giving up of their time.

But, I agree with @raisedadead and I'm sorry if I was off-topic. I felt that the suggestion might not be implemented fairly, and was trying the best I could to show a possible problem with giving badges to people on the forum as compared to people on other platforms.

**Comment by raisedadead** [comment]
Yes, that what I mean as well, by the metrics to be normalized. 

That is **just for the sake of an example**  20 points on chat can be equivalent of say one like or accepted solution on forum, or something on similar lines.

Long time members of the community can make a opinionated guess on what such a scale could be for such indicators? Don't you agree?

**Comment by IsaacAbrahamson** [comment]
>Yes, that what I mean as well, by the metrics to be normalized.
Ok, I understand better now.

> Long time members of the community can make a opinionated guess on what such a scale could be for such indicators? Don't you agree?

I agree. I didn't have the best success on Gitter and haven't been on there in a while, so I have no right to give an opinion, but I think that sounds like a good approach. My one problem as I mentioned above is how to classify something on the forum. Do we reward based on likes or replies, because both approaches may have problems identifying the post helpful campers.

**Comment by ojongerius** [comment]
Love this idea! Not a fan of voting, as mentioned before: easy to game, very subjective.

I'd suggest:
* gather metrics: brownie points, PRs raised, reviewed, lines deleted && lines added, forum activity and other things I've not thought off. I'm skeptical of lines of chat in Gitter.
* finish with the FCC team doing a sanity check

I'm sure Camperbot would not mind in assisting with gathering metrics and suggesting nominees for the team to check.

**Comment by QuincyLarson** [comment]
@raisedadead I like the idea of adding this. It would have to be after we ship beta.

As for whom to give it to, I propose this: @raisedadead decides. 

Having a single person make the decision makes this process simple. @raisedadead is highly active in a number of repositories, and himself is a prolific contributor. He would be a good judge of whether a particular contributor meets the "prolific contributor" status. 

I propose we call it "Super Contributor" and assign a year to it. Sort of like "Super Contributor 2018". 

This is similar to how Quora reward their contributors with a "Top Writer 2017" badge.

When @Raisedadead notices a contributor who's contributing a lot, he can at his discretion award them the "Super Contributor 20XX" badge and can publish a Medium article publicly trumpeting their contributions.
